### PR TITLE
windows supporting

### PR DESCRIPTION
**Pull request by cloudedcat**

### CODE REVIEW

File: udt.h [hunk 1]
```diff
@@ -80,14 +81,6 @@ written by
          // VC 6.0 does not support unsigned __int64: may cause potential problems.
          typedef __int64 uint64_t;
       #endif
-
```
Comment (Lupus) [inline udt.h]: I would suggest to add a `#ifdef` here instead of removing these lines and toggle it explicitly in cflags.
This way the udt sources can be build as a library if need be.

File: udt_base.h [hunk 2]
```diff
@@ -25,18 +26,9 @@
          // VC 6.0 does not support unsigned __int64: may cause potential problems.
          typedef __int64 uint64_t;
       #endif
-
```
Comment (Lupus) [inline udt_base.h]: Same for these changes

File: Makefile [hunk 3]
```diff
@@ -1,4 +0,0 @@
-go-deps:
```
Comment (Lupus) [inline Makefile]: This change is unrelated to Windows support, please file a separate PR for this.

File: udtcat/udtcat.go [hunk 4]
```diff
@@ -32,8 +32,8 @@ var Usage = func() {
 
 Usage:
 
-  listen: %s [<local address>] <remote address>
```
Comment (Lupus) [inline udtcat/udtcat.go]: This has nothing to do with Windows support as well, please move this into a separate commit and file a separate PR for this.

File: Makefile (hunk 3 above)
Comment (cloudedcat) [reply Makefile]: I came it back

File: udt.h (hunk 1 above)
Comment (cloudedcat) [reply udt.h]: ok. fixed

File: udt_base.h (hunk 2 above)
Comment (cloudedcat) [reply udt_base.h]: fixed

File: udtcat/udtcat.go (hunk 4 above)
Comment (cloudedcat) [reply udtcat/udtcat.go]: fixed

File: fd_rw.go [hunk 5]
```diff
@@ -8,14 +8,14 @@ import (
...
-// #cgo windows CFLAGS: -DERROR_NO_WINDOWS_SUPPORT
+// #cgo windows CXXFLAGS: -DWIN32 -DUDT_EXPORTS
+// #cgo windows LDFLAGS: -static-libgcc -static-libstdc++ -static -lkernel32 -luser32 -lws2_32
 // #cgo i386 CFLAGS: -DIA32
 // #cgo amd64 CFLAGS: -DAMD64
 // #cgo CFLAGS: -Wall -finline-functions -O3 -fno-strict-aliasing -fvisibility=hidden
 // #cgo LDFLAGS: -lstdc++ -lm
 // #include "udt_c.h"
 // #include <errno.h>
-// #include <arpa/inet.h>
```
Comment (Lupus) [inline fd_rw.go:18]: What's wrong with this comment? :)

**Comment by Lupus** [comment]
So it builds on windows with this small changeset?

File: fd_rw.go (hunk 5 above)
Comment (cloudedcat) [reply fd_rw.go:18]: Looks like unused library

**Comment by cloudedcat** [comment]
yes
